awswrangler/athena/_read.py

```diff
@@ -232,7 +234,14 @@ def _fetch_parquet_result(
     _logger.debug("metadata_path: %s", metadata_path)
     paths: List[str] = _extract_ctas_manifest_paths(path=manifest_path, boto3_session=boto3_session)
     if not paths:
-        return _empty_dataframe_response(bool(chunked), query_metadata)
+        if not temp_table_fqn:
+            raise exceptions.EmptyDataFrame("Query would return untyped, empty dataframe.")
```

Comment:
To get around the case where the temp_table_name is unavailable while reading from the cache is perhaps to *voluntarily* fail the query. Since [`_resolve_query_with_cache`](https://github.com/awslabs/aws-data-wrangler/blob/7b9bc57de9d9b8da80534ebd261a6e45abb7b787/awswrangler/athena/_read.py#L807) is wrapped in a try, it will default to `_resolve_query_without_cache` and create a new temp table

Reply:
Sounds perfect to me. Thank you for the solution 🙂